src/main/java/org/springframework/data/neo4j/core/mapping/Neo4jMappingContext.java

```diff
@@ -247,13 +253,45 @@ protected Neo4jPersistentProperty createPersistentProperty(Property property, Ne
 	@Override
 	public Optional<Neo4jPersistentEntity<?>> addPersistentEntity(TypeInformation<?> typeInformation) {
 
-		NodeDescription<?> existingDescription = this.getNodeDescription(typeInformation.getRawTypeInformation().getType());
+		NodeDescription<?> existingDescription = this.doGetPersistentEntity(typeInformation);
 		if (existingDescription != null) {
 			return Optional.of((Neo4jPersistentEntity<?>) existingDescription);
 		}
 		return super.addPersistentEntity(typeInformation);
 	}
 
+	/**
+	 * @param typeInformation The type to retrieve an persistent entity for
+	 * @return An optional persistent entity
+	 * @see #doGetPersistentEntity(Class)
+	 */
+	private @Nullable Neo4jPersistentEntity<?> doGetPersistentEntity(TypeInformation<?> typeInformation) {
```

Comment:
For this method and the next: Can we have the `@Nullable` above the declaration, please.